src/kOS/Communication/MessageStructure.cs

```diff
@@ -41,10 +42,22 @@ private void InitializeSuffixes()
         {
             AddSuffix("SENTAT", new Suffix<kOS.Suffixed.TimeSpan>(() => new kOS.Suffixed.TimeSpan(Message.SentAt)));
             AddSuffix("RECEIVEDAT", new Suffix<kOS.Suffixed.TimeSpan>(() => new kOS.Suffixed.TimeSpan(Message.ReceivedAt)));
-            AddSuffix("SENDER", new Suffix<VesselTarget>(() => Message.Vessel));
+            AddSuffix("SENDER", new Suffix<VesselTarget>(GetVesselTarget));
             AddSuffix("CONTENT", new Suffix<Structure>(DeserializeContent));
         }
 
+        public VesselTarget GetVesselTarget()
+        {
+            Vessel vessel = FlightGlobals.Vessels.Find((v) => v.id.ToString().Equals(Message.Vessel));
+
+            if (vessel == null)
+            {
+                throw new KOSCommunicationException("Vessel does not exist");
+            }
+
+            return new VesselTarget(vessel, shared);
+        }
+
```

Comment:
That might be better

``if msg:sender:istype("Vessel") { stuff... }``

With the documentation for the vessel suffix saying "If the vessel stopped existing after the message was sent (i.e. it crashed or was recovered), then the type of the sender suffix won't be ``Vessel``.  So you can use the test (show the above example) to check if the vessel still exists.

That way we don't explicitly say the result is boolean false even though it is.  My reason for this is that I don't like the notion of a boolean that can never be true and only is of type boolean when it's false.  (One of the good features that was lost when going from old C to modern languages was the ability to treat *everything* as a boolean, which often allowed this kind of check to always work - null is false, non-null is true).